mustermann/lib/mustermann/ast/expander.rb

```diff
@@ -12,11 +12,21 @@ module AST
     class Expander < Translator
       raises ExpandError
 
-      translate Array do |*args|
-        inject(t.pattern) do |pattern, element|
-          t.add_to(pattern, t(element, *args))
+      # :nocov:
+      if RUBY_VERSION < "3"
+        translate Array do |*args|
+          inject(t.pattern) do |pattern, element|
+            t.add_to(pattern, t(element, *args))
+          end
+        end
+      else
+        translate Array do |*args, **kwargs|
+          inject(t.pattern) do |pattern, element|
+            t.add_to(pattern, t(element, *args, **kwargs))
+          end
         end
       end
```

Comment:
Another possibility would be to call `Proc#ruby2_keywords` on this block which delegates `*args` (and other blocks doing the same).
That seems the cleanest solution.